### PR TITLE
Big fire rework

### PR DESCRIPTION
**Pull request by Coolthulhu**
As the 3D fields are very near (I'm going to fix that issue with sound markers in a moment), I decided to optimize fires. Current fire code is painful when a small house burns, so it is obviously not fit to simulate 3 story building burning.

Here I rework the fire-related functions and heavily alter the fire mechanics. The new fire is obviously much faster, but also:
- The medium fire stage happens much more often. It isn't just a rare transient stage between small and raging fire that lasts 3 turns - now most fires will only grow to medium size.
- Fires are more chaotic, spread earlier (from medium stage) and more randomly
- Fires spread much, much slower. In fact, it is still possible (if dangerous) to loot a burning house. In master branch fire can easily catch up with a healthy zombie running in front of it. Here they creep rather than explode.
- Fires "exchange fuel" more readily, resulting in them forming blobs or pockets of fire rather single tiles.
- Fires smoke less. This was necessary, as smoke was a big part of the reason why is fire so slow. In order to compensate, large fires aren't fully transparent.
- Fires produce less hot air. Hot air is a gas and thus causes slowdown. To compensate, fire distance is not squared for temperature calculations. This makes raging fires more dangerous to be around.
- Buffed fire's ability to consume flesh. Butchering corpses is very easy and funeral pyres were very ineffective - now they're closer to pulping in terms of time efficiency (assuming a big pile of corpses).
- Weakened effects of rain on fires. The new nerfed fire spreads more and so rain affects it more.

This makes fire less dependable (especially without fuel), more chaotic, but also less straining on the CPU.

Includes #12326

### CODE REVIEW

**Comment by Zireael07** [comment]
So is it still safe to light withered plants or sticks to get small fires?

**Comment by Coolthulhu** [comment]
Yes, it's just that they'll die faster unless you fuel them.

File: src/field.cpp [hunk 1]
```diff
@@ -479,6 +428,114 @@ If you need to insert a new field behavior per unit time add a case statement in
 bool map::process_fields_in_submap( submap *const current_submap,
                                     const int submap_x, const int submap_y, const int submap_z )
 {
+    const auto get_neighs = [this]( const tripoint &pt ) {
```
Comment (kevingranade) [inline src/field.cpp]: Please either change this to "neighbors", or use it to retrieve horse sound effects, I'm fine with either one ;)

**Comment by Coolthulhu** [comment]
Removed horse sounds. Let's hope that won't make it less stable.

**Comment by Rivet-the-Zombie** [comment]
>Removed horse sounds. Let's hope that won't make it less stable.

If we keep horsin' around like this they'll put us out to pasture.

**Comment by DavidKeaton** [comment]
What an odd requestrian topic this trotted to. Don't hoof a hard time trying to horseshoe these fixes in if it's stallion the PR!

**Pull request merged**